### PR TITLE
[SQLite] Always use foreign keys

### PR DESCRIPTION
**Pull request by natemcmaster**
Always calls `PRAMGA foreign_keys = ON;` after opening a new connection.

### CODE REVIEW

File: src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs [hunk 1]
```diff
@@ -20,5 +22,25 @@ public SqliteDatabaseConnection([NotNull] IDbContextOptions options, [NotNull] I
...
+            base.Open();
+            EnableForeignKeys();
+        }
+        public override async Task OpenAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await base.OpenAsync(cancellationToken);
+            EnableForeignKeys();
+        }
+
+        private void EnableForeignKeys()
```
Comment (bricelam) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: Need a switch to turn this off too.

Comment (bricelam) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: Possibly similar to SQL Server's [SuppressAmbientTransactionWarning](https://github.com/aspnet/EntityFramework/blob/dev/src/EntityFramework.SqlServer/Extensions/SqlServerDbContextOptionsBuilder.cs#L52).

**Comment by natemcmaster** [comment]
@bricelam updated. Can be switched off by using `.UseSqlite().SuppressForeignKeysEnforcement()`

**Comment by ErikEJ** [comment]
:thumbsup:

File: src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs [hunk 2]
```diff
@@ -12,13 +15,45 @@ namespace Microsoft.Data.Entity.Sqlite
 {
     public class SqliteDatabaseConnection : RelationalConnection
     {
+        private readonly bool _enforceForeignKeys = true;
```
Comment (bricelam) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: Is caching this an issue? Would the user want to change it over the lifetime of this instance? E.g. turn it off for seed data, but turn it on for the rest of the app's operations.

File: test/EntityFramework.Sqlite.FunctionalTests/ForeignKeyTest.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        public ForeignKeyTest()
+        {
+            _testStore = SqliteTestStore.CreateScratch();
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_enforces_foreign_key(bool suppress)
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/ForeignKeyTest.cs]: These tests could be simplified to just assert that `PRAGMA FOREIGN_KEYS;` returns `0` or `1`.

File: src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs (hunk 2 above)
Comment (natemcmaster) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: This is cached, but only affects connection EF opens. For someone seeding data, they could turn it off by executing a pragma statement.

Comment (bricelam) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: Maybe SuppressForeignKeysEnforcement is YAGNI. We should look at what it's like to suppress it globally and one-off and discuss with others.

Comment (bricelam) [inline src/EntityFramework.Sqlite/SqliteDatabaseConnection.cs]: Regardless, it looks like we don't expose a way of changing DbContextOptions over the lifetime of an instance, so caching is fine for now.

**Comment by bricelam** [comment]
:shipit:

**Pull request merged**